src/claim.lua

```diff
@@ -1,27 +1,54 @@
 SetHeader("Access-Control-Allow-Origin", "*")
 
+if GetMethod() ~= 'GET' and GetMethod() ~= 'POST' and GetMethod() ~= 'HEAD' then
+    Log(kLogWarn, "got %s request from %s" % {GetMethod(), FormatIp(GetRemoteAddr() or "0.0.0.0")})
+    ServeError(405)
+    SetHeader('Allow', 'GET, POST, HEAD')
+    return
+end
+
 local ip = GetRemoteAddr()
+if not ip then
+    SetStatus(400, "IPv4 Games only supports IPv4 right now")
+    Write("IPv4 Games only supports IPv4 right now")
+    return
+end
+
 local ip_str = FormatIp(ip)
 local name = GetParam("name")
 local escaped_name = EscapeHtml(name)
 
 if name == nil or name == "" then
-    return ServeError(400, "Name query param was blank")
+    SetStatus(400, "Name query param was blank")
+    Write("Name query param was blank")
+    return
 elseif #name > 40 then
-    return ServeError(400, "Name must be no more than 40 characters")
+    SetStatus(400, "name must be no more than 40 characters")
+    Write("name must be no more than 40 characters")
+    return
 else
     local invalid_char = re.search("[^!-~]", name)
     if invalid_char ~= nil then
-        return ServeError(400, string.format("Invalid character in name: \"%s\"", invalid_char))
+        Log(kLogWarn, string.format("Invalid character in name: \"%s\"", invalid_char))
+        return ServeError(400)
```

Comment:
Should probably still provide an error string here even if we don't want to reflect the char, like "Name contains invalid character at index %d"